Query features with arcade expression (#74)
Esri/arcgis-maps-sdk-kotlin-samples: query-features-with-arcade-expression/README.md

# Query features with arcade expression

Query features on a map using an Arcade expression.

![QueryFeaturesWithArcadeExpression](query-features-with-arcade-expression.png)

## Use case

Arcade is a portable, lightweight, and secure expression language used to create custom content in ArcGIS applications. Like other expression languages, it can perform mathematical calculations, manipulate text, and evaluate logical statements. It also supports multi-statement expressions, variables, and flow control statements. What makes Arcade particularly unique when compared to other expression and scripting languages is its inclusion of feature and geometry data types. This sample uses an Arcade expression to query the number of crimes in a neighborhood in the last 60 days.

## How to use the sample

> Tap on any neighborhood to see the number of crimes in the last 60 days in a callout.

Reviewer: ```suggestion
Tap on any neighborhood to see the number of crimes in the last 60 days in a TextView.
```

## How it works

1. Create a `PortalItem` using the URL and ID.

> 2. Create a `ArcGISMap` using the portal item.

Reviewer 2: ```suggestion
2. Create an `ArcGISMap` using the portal item.
```

3. Set up a listener for taps on the map.
4. Identify the visible layer where it is tapped using `mapView.identifyLayer()` and get the feature.
5. Create the following `ArcadeExpression`:

   ```kotlin

>    expressionValue = "var crimes = FeatureSetByName(\$map, 'Crime in the last 60 days');\n"
>     "return Count(Intersects(\$feature, crimes));"

Reviewer: ```suggestion
expressionValue = "var crimes = FeatureSetByName(\$map, 'Crime in the last 60 days');\n" +
    "return Count(Intersects(\$feature, crimes));"
```

Author: I believe this indentation is needed, otherwise the code snippet is not rendered since its inside a list.

Reviewer: Okay, but will still need a `+` between the two quotes.

Author: Ah, thanks I missed that.

   ```

> 6. Create an `ArcadeEvaluator` using the Arcade expression and `ArcadeProfile.FORM_CALCULATION`.

Reviewer: ```suggestion
6. Create an `ArcadeEvaluator` using the Arcade expression and `ArcadeProfile.FormCalculation`.
```
